src/Symfony/Component/Serializer/Context/Encoder/CsvEncoderContextBuilder.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Context\Encoder;
+
+use Symfony\Component\Serializer\Context\AbstractContextBuilder;
+use Symfony\Component\Serializer\Encoder\CsvEncoder;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+/**
+ * A helper providing autocompletion for available CsvEncoder options.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ */
+final class CsvEncoderContextBuilder extends AbstractContextBuilder
+{
+    /**
+     * Configures the column delimiter character.
+     *
+     * Must be a single character.
+     *
+     * @throws InvalidArgumentException
+     */
+    public function withDelimiter(?string $delimiter): static
```

Comment:
Maybe could you use the `non-empty-string` Pslam type? Same for all other 1 char length strings. (nitpicking)